infrastructure/builder/src/build/monoimage.js

```diff
@@ -147,8 +156,54 @@ const generateMonoimageTasks = ({tasks, baseDir, spec, cfg, cmdOptions}) => {
     },
   });
 
-  // TODO: add hooks push for web-ui
-  // TODO: add entry point for docker run for web-ui
+  hooks.push({
+    name: 'web-ui',
+    build: async (requirements, utils, procs) => {
+      const cacheDir = path.join(workDir, 'ui/cache');
+      if (!fs.existsSync(cacheDir)) {
+        fs.mkdirSync(cacheDir);
+      }
+
+      utils.step({title: 'Run Yarn Install'});
+
+      await dockerRun({
+        image: nodeImage,
+        workingDir: '/app/ui',
+        env: ['YARN_CACHE_FOLDER=/ui/cache'],
+        command: ['yarn'],
```

Comment:
Will this work with `--production` to avoid devDependencies?

Reply:
> Running yarn with no command will run yarn install, passing through any provided flags.

From the docs

Reply:
Yes, and adding `--production` will cause it to skip installing any `devDependencies`, which should save some space in the docker iamge.

Reply:
Why don't we add `--production` anywhere? I was simply copying your code

Reply:
I think there's a bug to add that.  I tried it once and it didn't work, and I didn't try too hard to figure out why.  If it doesn't work for tc-ui, we can file another bug and leave out `--production` for now.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1521648

If it's too much trouble, we can drop this and just add tc-ui to that bug.

Reply:
I see. It sounded as if there is `--production` everywhere, and I just missed it for some reason :)

Reply:
I tried both `'yarn', '--production'` and `'yarn', 'install', '--production'` - I still see things like `@neutrinojs/karma@9.0.0-beta.1` in the logs (that's a dev dependency)

Reply:
OK, let's worry about it later..